src/main/scala/scatan/model/game/Game.scala

```diff
@@ -0,0 +1,150 @@
+        GameMap(2),
+        Building.empty(players),
+        ResourceCard.empty(players),
+        DevelopmentCardsOfPlayers.empty(players),
+        Award.empty()
+      )
+
+  def apply(
+      players: Seq[Player],
+      gameMap: GameMap,
+      buildings: Buildings,
+      resourceCards: ResourceCards,
+      developmentCardsOfPlayers: DevelopmentCards
+  ): Game =
+    GameImpl(players, gameMap, buildings, resourceCards, developmentCardsOfPlayers)
+
+private final case class GameImpl(
+    players: Seq[Player],
+    gameMap: GameMap,
+    buildings: Buildings,
+    resourceCards: ResourceCards,
+    developmentCards: DevelopmentCards,
+    assignedAwards: Awards = Award.empty()
+) extends Game:
+
+  def awards: Awards =
+    val precedentLongestRoad = assignedAwards(Award(AwardType.LongestRoad))
+    val longestRoad =
+      buildings.foldLeft(precedentLongestRoad.getOrElse((Player(""), 0)))((playerWithLongestRoad, buildingsOfPlayer) =>
+        val roads = buildingsOfPlayer._2.filter(_.buildingType == BuildingType.Road)
+        if roads.sizeIs > playerWithLongestRoad._2 then (buildingsOfPlayer._1, roads.size)
+        else playerWithLongestRoad
+      )
+    val precedentLargestArmy = assignedAwards(Award(AwardType.LargestArmy))
+    val largestArmy =
+      developmentCards.foldLeft(precedentLargestArmy.getOrElse(Player(""), 0))((playerWithLargestArmy, cardsOfPlayer) =>
+        val knights = cardsOfPlayer._2.filter(_.developmentType == DevelopmentType.Knight)
+        if knights.sizeIs > playerWithLargestArmy._2 then (cardsOfPlayer._1, knights.size)
+        else playerWithLargestArmy
+      )
+    Map(
+      Award(AwardType.LongestRoad) -> (if longestRoad._2 >= 5 then Some((longestRoad._1, longestRoad._2))
+                                       else precedentLongestRoad),
+      Award(AwardType.LargestArmy) -> (if largestArmy._2 >= 3 then Some((largestArmy._1, largestArmy._2))
+                                       else precedentLargestArmy)
+    )
+
+  private def verifyResourceCost(player: Player, cost: Cost): Boolean =
+    cost.foldLeft(true)((result, resourceCost) =>
+      result && resourceCards(player).count(_.resourceType == resourceCost._1) >= resourceCost._2
+    )
+
+  override def assignBuilding(building: Building, player: Player): Game =
+    this.copy(
+      buildings = buildings.updated(player, buildings(player) :+ building),
+      assignedAwards = awards
+    )
+
+  override def build(building: Building, player: Player): Game =
+    if verifyResourceCost(player, building.buildingType.cost) then
+      val remainingResourceCards = building.buildingType.cost.foldLeft(resourceCards(player))((cards, resourceCost) =>
+        cards.filter(_.resourceType != resourceCost._1).drop(resourceCost._2)
+      )
+      val gameWithConsumedResources = this.copy(resourceCards = resourceCards.updated(player, remainingResourceCards))
+      gameWithConsumedResources.assignBuilding(building, player)
+    else this
```

Comment:
The difference between this two methods is that the `assignBuilding` doesn't check the actual resource avaiability necessary to build the construction, it's useful for the INITIAL PLACEMENT PHASE and it semplify Unit Testing.